src/replication.c

```diff
@@ -4064,7 +4064,7 @@ int cancelReplicationHandshake(int reconnect) {
 }
 
 /* Set replication to the specified primary address and port. */
-void replicationSetPrimary(char *ip, int port, int full_sync_required) {
+void replicationSetPrimary(char *ip, int port, int full_sync_required, int disconnect_blocked) {
```

Comment:
We are trying to get rid of boolean integers. can we make it a bool?

Reply:
It was decided here:

* #1278

We allow bool now but we don't require to change old code to bool. I guess it's also OK to keep the old style when changing old code (such as here). I don't have a strong opinion though.